corehq/apps/data_interfaces/tests/test_deduplication_rules.py

```diff
@@ -0,0 +1,94 @@
+import json
+from unittest.mock import Mock, patch
+
+from django.test import RequestFactory, TestCase
+
+from corehq.apps.data_interfaces.models import AutomaticUpdateRule
+from corehq.apps.data_interfaces.views import (
+    DeduplicationRuleCreateView,
+    DeduplicationRuleEditView,
+    HttpResponseRedirect,
+)
+
+
+class DeduplicationRuleCreateViewTest(TestCase):
```

Comment:
A good additional test could be checking what happens when a rule is created with the same name as another rule on that domain (basically checking what happens when rule=None in validate_rule_params).

Reply:
Added test for create view